core/modules/wsched/BlendScheduler.cc

```diff
@@ -67,32 +68,59 @@ BlendScheduler* dbgBlendScheduler=nullptr; ///< A symbol for gdb
 ////////////////////////////////////////////////////////////////////////
 // class BlendScheduler
 ////////////////////////////////////////////////////////////////////////
+/// @param scanScheduler - schedulers must be listed in ascending priority.
 BlendScheduler::BlendScheduler(std::string const& name,
                                int schedMaxThreads,
                                std::shared_ptr<GroupScheduler> const& group,
-                               std::shared_ptr<ScanScheduler> const& scanFast,
-                               std::shared_ptr<ScanScheduler> const& scanMedium,
-                               std::shared_ptr<ScanScheduler> const& scanSlow)
-    : SchedulerBase{name, 0, 0}, _schedMaxThreads{schedMaxThreads},
-      _group{group}, _scanFast{scanFast}, _scanMedium{scanMedium}, _scanSlow{scanSlow} {
+                               std::vector<std::shared_ptr<ScanScheduler>> const& scanSchedulers)
+    : SchedulerBase{name, 0, 0, 0}, _schedMaxThreads{schedMaxThreads},
+      _group{group}, _scanFast{scanSchedulers[0]} {
```

Comment:
What happens if `scanSchedulers` is empty? At minimum replace `[0]` with `.at(0)`